src/super_gradients/training/utils/detection_utils.py

```diff
@@ -752,6 +752,199 @@ def crowd_ioa(det_box: torch.Tensor, crowd_box: torch.Tensor) -> torch.Tensor:
     return inter / det_area[:, None]  # crowd_ioa = inter / det_area
 
 
+class DetectionMatching(ABC):
```

Comment:
Please add missing docstring to this class (brief description of the class goal) and to each abstract method (Specifically what are the expected return types are)

Reply:
Added